boring/src/pkey.rs

```diff
@@ -222,6 +222,26 @@ where
     {
         unsafe { ffi::EVP_PKEY_cmp(self.as_ptr(), other.as_ptr()) == 1 }
     }
+
+    /// Outputs a copy of the "raw" form of the public key. Only supported for certain key types.
+    ///
+    /// Returns the number of bytes written, or the size of the raw form if `out` is empty.
+    #[corresponds(EVP_PKEY_get_raw_public_key)]
+    pub fn raw_public_key(&self, out: &mut [u8]) -> Result<usize, ErrorStack> {
```

Comment:
I would rather have two methods, `raw_public_key` and `raw_public_key_len`, rather than piggyback `raw_public_key` to do a different thing when `out` is empty.

Reply:
With 2 separate methods, you can change `raw_{public,private}_key` to the following signature:

```rust
fn raw_public_key<'a>(&self, out: &'a mut [u8]) -> Result<&[u8], ErrorStack> { … }
```